### ISSUE TITLE
Use Cross Entropy Loss for MNIST

### ISSUE THREAD

**Issue by pranav-ap** [pranav-ap/bertelsmann-scholarship#1]

**Comment by pranav-ap** [comment]
The negative log-likelihood loss function is often used in combination with a Softmax activation function to define how well your neural network classifies data.

NLL loss function is defined as loss=-log(y).

- produces a high value when the values of the output layer are evenly distributed and low. 
- produces relative high values when the classification is wrong
- When the value of the output layer matches that of the expected value, it produces a very low value.

**Comment by pranav-ap** [comment]
**Shannon Entropy** https://www.youtube.com/watch?v=9r7FIXEAGvs
